### ISSUE TITLE
Please guide in rendering the mesh?

### ISSUE THREAD

**Issue by icarito** [mapbox/martini#5]
I've tried different attempts using https://github.com/andreasplesch/aframe-triangleset-component but I get this from the Mount Fuji 256 data:

![imagen](https://user-images.githubusercontent.com/199755/64091631-ac960c00-cd16-11e9-9f92-df60260daba5.png)
Which doesn't look to me like the input image:
![fuji256](https://user-images.githubusercontent.com/199755/64091663-d8b18d00-cd16-11e9-8463-a3033f07da30.png)

Mainly because of the pointy spikes. 
I'm using 
`<a-triangleset projectdir="z" material="roughness: 0.2; metalness: 0.3; color: green" rotation="90"  position="-60 -100 30" scale="1 1 1" :vertices="this.vertices" :triangles="this.triangles"></a-triangleset>`

Where this.vertices and this.triangles come from a Vue component that uses the data from tile.getMesh(80) to make these attribute data.

Please advise what am I doing wrong?
Thanks for your guidance.

**Comment by icarito** [comment]
![imagen](https://user-images.githubusercontent.com/199755/64091990-79ed1300-cd18-11e9-8ba6-1dd237f1948b.png)
Here's my result when using getMesh(30).

**Comment by icarito** [comment]
@andreasplesch maybe this is a question for you? ;-)
I also tried rendering with pythreejs but found similar issues.
![imagen](https://user-images.githubusercontent.com/199755/64092260-a190ab00-cd19-11e9-94da-936f9d2ae97f.png)
Obviously I'm a beginner at 3D!
Thanks in advance for you kind advice.

**Comment by andreasplesch** [comment]
Please provide a live example, on glitch, jsfiddle,

**Comment by icarito** [comment]
Thanks @andreasplesch - I should've done from the beginning but I have a backend in Python3.7 that is not compatible with Glitch's older interpreter.
I've implemented a new project doing only this:
https://glitch.com/~mahogany-eggplant

My loose aim is to make a libre ("open source") videogame / interactive-experience about the Amazonian rainforest (where I live). Thank you for your nice components!

**Comment by mourner** [comment]
This is likely because the vertex data generated by this library is `[x, y]` pairs, without the `z` component (assuming the renderer will look these up through a texture). I probably need to add more docs for this, and maybe add an option to add `z` to the buffers too.

**Comment by andreasplesch** [comment]
Yes, the z component is expected to be provided with the x and y. It would be not too hard to inject the z value from the data into the returned vertices array with something like:
```this.vertices=[];
for (i=0, j=0; i<mesh.vertices.length/2; i++) {
  let  x = this.vertices[i*2], y = this.vertices[i*2+1];
  this.vertices.push(x);
  this.vertices.push(response.data[y*257+x]/100);
  this.vertices.push(y);
}
```
Typically, some processing has to be done anyways after the meshing, the returned x and y coordinates are just grid coordinates, not spatial coordinates.
[The up component is typically y]
[Also, I am not sure how well the component still works with current three and aframe]

**Comment by icarito** [comment]
Thanks @mourner and @andreasplesch !
![imagen](https://user-images.githubusercontent.com/199755/64181670-8b274400-ce2c-11e9-98ef-b0d23b602b8a.png)
That was the missing insight. I did a little tweaking to your snippet and it worked wonderfully!
Thank you  both for the Library and the Component (it works fine with AFrame 0.9.2)
Here's the code and test:
https://glitch.com/~mahogany-eggplant
https://mahogany-eggplant.glitch.me/

@mourner Yes a little more docs would be great, I loved the [demo](https://observablehq.com/@mourner/martin-real-time-rtin-terrain-mesh) but couldn't figure it out.

Thanks!
:heart:

**Comment by arpu** [comment]
@icarito  your glitch test is deleted?

**Comment by icarito** [comment]
@arpu sorry I just renamed to a more accurate name!

https://glitch.com/~mount-fuji

**Comment by arpu** [comment]
@icarito  thx for poinzing me to the new glitch
but i get 
three.js:11050 THREE.DirectGeometry: Faceless geometries are not supported.

and nothing is shown

**Comment by icarito** [comment]
The mesh can be seen at the bottom, I didn't center it or anything - look to the bottom left

**Comment by icarito** [comment]
bottom right I mean, sorry it's not in front of the camera

**Comment by arpu** [comment]
ahh :> thx, great example!

**Comment by mourner** [comment]
Let's reopen the issue until we have a proper demo / better docs.

**Comment by arpu** [comment]
@mourner  testing with aframe master and THREEjs v108 and BufferGeometry
https://codesandbox.io/s/aframe-draco-gltf-model-plus-77rld


![Bildschirmfoto von 2019-09-19 14-08-23](https://user-images.githubusercontent.com/14173/65243388-8cbd6100-dae8-11e9-9777-d7d8b92d467d.png)

**Comment by arpu** [comment]
some update
https://codesandbox.io/s/aframe-draco-gltf-model-plus-o7zki

![Bildschirmfoto von 2019-09-19 16-08-20](https://user-images.githubusercontent.com/14173/65251513-bf228a80-daf7-11e9-9b0c-97048a0fba44.png)

**Comment by arpu** [comment]
@mourner  what to you think on return the indices on getMesh ? 

than the BufferGeometry could be set with setIndex()
https://threejs.org/docs/#api/en/core/BufferGeometry.setIndex

**Comment by mourner** [comment]
@arpu that's what `triangles` property in the returned object contains.

**Comment by arpu** [comment]
awsome works! i update the codepen on https://codesandbox.io/s/aframe-draco-gltf-model-plus-o7zki

**Comment by arpu** [comment]
@mourner one more question how can set the createTile with a BufferGeometry? 

the idea is bevore i use Martini i use 

```
  var geometry = new THREE.Geometry().fromBufferGeometry(this.groundGeometry);
    geometry.mergeVertices();
    var modifier = new THREE.SubdivisionModifier(1);
    var smoothGeometry = modifier.modify(geometry);
    if (this.groundGeometry) this.groundGeometry.dispose();
    this.groundGeometry = new THREE.BufferGeometry().fromGeometry(smoothGeometry);
```

to get a smoother Terrain bevore Martini
